jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanPropertyIntrospector.java

```diff
@@ -146,6 +153,19 @@ private static void _introspect(Class<?> currType, Map<String, PropBuilder> prop
                         _propFrom(props, name).withIsGetter(m);
                     }
                 }
+                else if (isFieldNameGettersEnabled) {
+                    // This will allow getters with field name as their getters, like the ones generated by Groovy
+                    // If method name matches with field name, & method return type matches with field type
+                    // only then it can be considered a direct name getter.
+                    final Map<String, Field> fieldNameMap =
+                            stream(currType.getDeclaredFields()).collect(toMap(Field::getName, identity()));
```

Comment:
No I meant to pre-create lookup map before looping since set of Fields does not vary by method.

Reply:
done!